### PR TITLE
Adding Azure Game Developer Virtual Machine bicep template

### LINKED ISSUE #129: [Module Proposal]: Game Developer Virtual Machine
### Have you checked this module does not already exist in the registry?

Yes

### Why is the module needed?

We would like to simplify our template [here](https://github.com/Azure/azure-quickstart-templates/tree/master/application-workloads/azure-gamedev/gamedev-vm) by using a Bicep module. We will then go back and replace the Template with a direct call to the public module.

### Module path

modules/azure-gaming/game-dev-vm

### Describe the module

This Bicep module makes it easier for users to leverage the Game Dev Virtual Machine on a VM via ARM.
https://docs.microsoft.com/en-us/gaming/azure/game-dev-virtual-machine/

### PR DESCRIPTION
**Pull request by dciborow**
## Description

closes #129

If you haven't already, read the full [contribution guide](../CONTRIBUTING.md). The guide may have changed since the last time you read it, so please double-check. Once you are done and ready to submit your PR, edit the PR description and run through the relevant checklist below.

## Adding a new module

- [x] A proposal has been submitted and approved.
- [x] I have included "Closes #{module_proposal_issue_number}" in the PR description.
- [x] I have run `brm validate` locally to verify the module files.
- [x] I have run deployment tests locally to ensure the module is deployable.

## Updating an existing module

- [ ] This is a bug fix:
  - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
  - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
- [ ] I have run `brm validate` locally to verify the module files.
- [ ] I have run deployment tests locally to ensure the module is deployable.
- [ ] I have read the [Updating an existing module](../CONTRIBUTING.md#updating-an-existing-module) section in the contributing guide and updated the `version.json` file properly:
  - [ ] The PR contains backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`.
  - [ ] The PR contains backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] The PR contains breaking changes, and I have bumped the MAJOR version in `version.json`.

### CODE REVIEW

**Comment by dciborow** [comment]
#129

File: modules/azure-gaming/game-dev-vm/test/main.test.bicep [hunk 1]
```diff
@@ -0,0 +1,41 @@
+param location string = resourceGroup().location
+param vmName string = 'bicep'
+param adminName string = 'dcibadmin'
+param vmSize string = 'Standard_D14_v2'
```
Comment (dciborow) [inline modules/azure-gaming/game-dev-vm/test/main.test.bicep:4]:
```suggestion
param vmSize string = 'Standard_D4s_v3'
```

**Comment by dciborow** [comment]
@shenglol , this is also ready for review.

@alex-frankel , @bmoore-msft , there was a discussion about a "parameters" file for bicep which I jumped in on.

Take a look at this PR, and all of the parameters in my main.bicep. The majority of these are duplicates with the "GameDev VMSS" template. It would be ideal if there was a way that I could "add" parameter sets to a main.bicep, which are declared in a common folder. Scala has a process for this called "Traits", which is a little less then full on inheritance but I think it's a great example to look at.

File: modules/azure-gaming/game-dev-vm/README.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+| :-------- | :----: | :--------------------------------------- |
+| Host_Name | string | Game Developer Virtual Machine Host Name |
+| UserName  | string | Game Developer Virtual Machine User Name |
+
+## Examples
+
+### Using the Azure Game Developer VM module
+
+```bicep
+module gameDevVM'br/public:azure-gaming/game-dev-vm:1.0.0' = {
```
Comment (shenglol) [inline modules/azure-gaming/game-dev-vm/README.md:74]: The patch version number always starts at 1.
```suggestion
module gameDevVM 'br/public:azure-gaming/game-dev-vm:1.0.1' = {
```

File: modules/azure-gaming/game-dev-vm/test/main.test.bicep [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    vmName    : vmName
+    adminName : adminName
+    adminPass : adminPass
+    osType    : 'win10'
+    gameEngine: engine
+    vmSize    : vmSize
+  }
+}]
+
+// @batchSize(1)
```
Comment (shenglol) [inline modules/azure-gaming/game-dev-vm/test/main.test.bicep:29]: Is the test needed? If not, can we remove it?

**Comment by alex-frankel** [comment]
> It would be ideal if there was a way that I could "add" parameter sets to a main.bicep, which are declared in a common folder.

This is definitely a scenario we are interested in and something we are considering now that we have native bicep parameters and is called out in the spec. I hope we get to it soon!

File: modules/azure-gaming/game-dev-vm/test/main.test.bicep (hunk 3 above)
Comment (dciborow) [reply modules/azure-gaming/game-dev-vm/test/main.test.bicep:29]: @shenglol , in order for the rest of these tests to pass, I would need you to approve the rest of the VM images. 

How many images should we test out of the 6 total

Comment (dciborow) [reply modules/azure-gaming/game-dev-vm/test/main.test.bicep]: lets circle pack in a new PR though if we want to add more tests. going to clean this up for now.

**Review by shenglol — APPROVED** [review]
Looks good to me. @alex-frankel to take a look as well.

**Review by alex-frankel — APPROVED** [review]

**Pull request merged**